src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,137 @@
+    private final OrderItemRepository orderItemRepository;
+    private final ShoppingCartService shoppingCartService;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    public List<OrderResponseDto> getAllOrders(Long userId, Pageable pageable) {
+        return orderRepository.findByUserId(userId, pageable).stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    @Transactional
+    public OrderResponseDto placeOrder(CreateOrderRequestDto requestDto, User user) {
+        ShoppingCart shoppingCart = shoppingCartService.getShoppingCart(user.getId());
+
+        Order order = createOrder(user, requestDto);
+        Set<OrderItem> orderItems = getOrderItemsFromShoppingCart(shoppingCart, order);
+
+        order.setOrderItems(orderItems);
+        order.setTotal(calculateTotal(orderItems));
+        cartItemRepository.deleteAll(shoppingCart.getCartItems());
+
+        return orderMapper.toDto(orderRepository.save(order));
+    }
+
+    @Override
+    public List<OrderItemResponseDto> getOrderItems(Long userId, Long orderId, Pageable pageable) {
+        isOrderExists(userId, orderId);
+
+        return orderItemRepository.findByOrderId(orderId, pageable).stream()
+                .map(orderItemMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderItemResponseDto getOrderItem(Long userId, Long orderId, Long itemId) {
+        Order order = getOrder(userId, orderId);
+
+        OrderItem orderItem = order.getOrderItems().stream()
+                .filter(item -> item.getId().equals(itemId))
+                .findFirst()
+                .orElseThrow(
+                        () -> new EntityNotFoundException(
+                                "Can't find order item with id: " + itemId
+                                + " in order: " + orderId)
+                );
+        return orderItemMapper.toDto(orderItem);
+    }
+
+    @Override
+    public OrderResponseDto updateOrderStatus(Long orderId, UpdateOrderRequestDto requestDto) {
+        Order order = orderRepository.findById(orderId).orElseThrow(
+                () -> new EntityNotFoundException("Can't find order with id: " + orderId)
+        );
+
+        order.setStatus(requestDto.getStatus());
+        return orderMapper.toDto(orderRepository.save(order));
+    }
+
+    private Order getOrder(Long userId, Long orderId) {
+        return orderRepository.findByUserIdAndId(userId, orderId).orElseThrow(
+                () -> new EntityNotFoundException(
+                        "Can't find order with id: " + orderId + " for user: " + userId)
+        );
+    }
+
+    private void isOrderExists(Long userId, Long orderId) {
+        Order order = getOrder(userId, orderId);
+        if (order.getUser().getId().equals(userId)) {
+            throw new EntityNotFoundException(
+                    "Can't find order with id: " + orderId + " for user: " + userId);
+        }

```

Comment:
If you get order by userId it can not have userId not equal to userId in method argument, so no reason for this check